### PR TITLE
Adds Integration Sensor

### PR DESCRIPTION
**Pull request by dgomes**
**Description:**
The integration platform provides the Riemann sum (integration) of a numeric sensor.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#19703

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_components/sensor.energy.markdown [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+## {% linkable_title Configuration %}
+
+To enable Energy Sensor in your installation, add the following to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+sensor:
+  - platform: energy 
+    source: sensor.current_power
+    name: "energy consumed"
```
Comment (frenck) [inline source/_components/sensor.energy.markdown]: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

File: source/_components/sensor.energy.markdown [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    round: 2
```
Comment (frenck) [inline source/_components/sensor.energy.markdown]: Same as above.

File: source/_components/sensor.energy.markdown [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+{% configuration %}
+source:
+  description: The entity ID of the sensor providing power readings (must provide values in W or kW).
+  required: true
+  type: string
+name:
+  description: Name to use in the frontend.
+  required: false
```
Comment (frenck) [inline source/_components/sensor.energy.markdown]: Missing `default`

File: source/_components/sensor.energy.markdown [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  type: string
+round:
+  description: Round the calculated energy value to at most N decimal places.
+  required: false
```
Comment (frenck) [inline source/_components/sensor.energy.markdown]: Missing `default`

**Review by frenck — CHANGES_REQUESTED** [review]
Thanks for the PR @dgomes, I've reviewed it and left some small comments. Could you please take a look? Thanks 👍

File: source/_components/sensor.energy.markdown (hunk 3 above)
Comment (dgomes) [reply source/_components/sensor.energy.markdown]: done 👍

File: source/_components/sensor.energy.markdown (hunk 4 above)
Comment (dgomes) [reply source/_components/sensor.energy.markdown]: done 👍

File: source/_components/sensor.energy.markdown (hunk 1 above)
Comment (dgomes) [reply source/_components/sensor.energy.markdown]: Right!

**Comment by dgomes** [comment]
@frenck this sensor has changed scope after discussion in the original PR and therefore I've updated all the documentation

**Review by frenck — DISMISSED** [review]
Thanks for updating the PR @dgomes! 👍 

✅ Approved. Can be merged as soon as the parent PR gets merged.
⚠️ The `ha_release` might need an update before merging this one in.

**Review by frenck — APPROVED** [review]

**Comment by frenck** [comment]
Thanks, @dgomes! 👍

**Pull request merged**